src/main/java/carpet/commands/DrawCommand.java

```diff
@@ -32,38 +36,233 @@ public static void register(CommandDispatcher<ServerCommandSource> dispatcher)
+                                                    .executes((c) -> drawPyramid(c.getSource(),
+                                                        BlockPosArgumentType.getBlockPos(c, "center"),
+                                                        "circle",
+                                                        IntegerArgumentType.getInteger(c, "radius"),
+                                                        IntegerArgumentType.getInteger(c, "height"),
+                                                        BoolArgumentType.getBool(c,"pointing up?"),"z",
+                                                        BlockStateArgumentType.getBlockState(c, "block"), null)
+                                                    )
+                                                    .then(literal("replace")
+                                                        .then(argument("filter", BlockPredicateArgumentType.blockPredicate())
+                                                            .executes((c) -> drawPyramid(c.getSource(),
+                                                                BlockPosArgumentType.getBlockPos(c, "center"),
+                                                                "circle",
+                                                                IntegerArgumentType.getInteger(c, "radius"),
+                                                                IntegerArgumentType.getInteger(c, "height"),
+                                                                BoolArgumentType.getBool(c,"pointing up?"),"z",
+                                                                BlockStateArgumentType.getBlockState(c, "block"),
+                                                                BlockPredicateArgumentType.getBlockPredicate(c,"filter")
+                                                                )
+                                                            )
+                                                        )
+                                                    )
+                                                )
+                                            )
+                                        )
+                                    )
+                                )
+                            )
+                        )
+                        .then(literal("square")
+                            .then(argument("center", BlockPosArgumentType.blockPos())
+                                .then(argument("radius", IntegerArgumentType.integer(1))
+                                    .then(argument("height",IntegerArgumentType.integer(1))
+                                        .then(argument("pointing up?",BoolArgumentType.bool())
+                                            .then(literal("x")
+                                                .then(argument("block", BlockStateArgumentType.blockState())
```

Comment:
Might be useful to extract the <block> <replace <block>> subtree on its own branch so you dont need to repeat it 1000 times.

Reply:
ok lemme do dat 🧠

Reply:
ofc 🧠

Reply:
And now for something completely different